### PR TITLE
fix(gate): stop counting infra-errored samples as wrong answers

### PR DESCRIPTION
**Pull request by arthi-arumugam-git**
## What was wrong

The eval loop scores a sample `0.0` when a sandbox or system error exhausts its retries, and tags the record with `scoring_details.error_category` (`infra_error`, `system`) so it can be told apart from a wrong answer. Nothing downstream reads that tag. `gate.py` pairs those records like any other, so a sandbox outage on part of the candidate run reads as a batch of regressions: with the default policy, six infra-errored samples out of twenty against a clean baseline turn a GO into a NO-GO, and nothing in the report says why.

## Change

- `_metric_problem_values` skips records whose `error_category` is `infra_error` or `system`. A problem with no scored repeat left drops out of pairing on that arm, so both arms lose it.
- `BenchmarkGateResult` gains `n_infra_excluded_baseline` and `n_infra_excluded_candidate`, and a reason line is appended when either is non-zero, so the exclusion is visible in the JSON report.
- Graceful errors and solve timeouts are the model's own failures and still count as `0.0`, unchanged.

## Tests

`tests/test_gate.py`, in `TestGateVerdicts`:
- six candidate samples tagged `infra_error` out of twenty: verdict GO, `n_paired == 14`, `n_infra_excluded_candidate == 6`, reason present. Fails on `main`.
- the same six tagged `solve_timeout`: verdict stays NO-GO, nothing excluded.

`ruff check` and `ruff format` clean on both files. One pre-existing failure in this file on `main` in my environment, `TestClusteredCIIntegration::test_paired_delta_ci_with_clusters` (a width comparison), is unrelated and untouched.

## Not changed

The eval loop's own scoring of infra errors as `0.0` and the headline `pass@1` / `mean_reward` in the bundle are left as they are; this only stops the gate treating those samples as evidence of regression. If a policy-level knob is preferred over a fixed category set, happy to rework it that way.

## Summary by CodeRabbit

* **Bug Fixes**
  * Infrastructure and system-error samples are excluded from per-problem metrics and paired regression comparisons.
  * Model-attributable failures, including solve timeouts, remain included in benchmark results.
  * Bundle-level fallback scores are no longer used when exclusions leave insufficient paired evidence.
  * Benchmark reports now show baseline and candidate exclusion counts and explanatory reasons.

* **Configuration**
  * Infrastructure-error categories can now be configured globally or overridden for individual benchmarks.

### CODE REVIEW

File: src/nemo_evaluator/engine/gate.py [hunk 1]
```diff
@@ -57,6 +57,23 @@
...
 _SUPPORTED_GATED_METRICS = frozenset({"mean_reward", "pass@1"})
 
+# The eval loop scores a sample 0.0 when the *infrastructure* failed, not the
+# model: sandbox or system errors that exhausted their retries.  It tags those
+# records with ``scoring_details.error_category`` so they can be told apart
+# from a wrong answer.  Nothing downstream read that tag, so a sandbox outage in
+# the candidate run looked to the gate like a batch of wrong answers and could
+# turn a GO into a NO-GO.  These categories are excluded from paired gating;
+# model-attributable failures (graceful errors, solve timeouts) still count.
+_INFRA_ERROR_CATEGORIES = frozenset({"infra_error", "system"})
```
[the remark this answers is not in the document]
Comment (arthi-arumugam-git) [reply src/nemo_evaluator/engine/gate.py]: Fair point, and it matches the repo guideline. Moved the categories into the gate policy in ecd3ab5: `excluded_error_categories` on `defaults` and per benchmark, default `["infra_error", "system"]`, and `[]` switches the exclusion off. The gate reads it from the resolved policy now, so there is nothing hardcoded in gate.py any more. Also added a line for it in the quality-gate tutorial.

File: src/nemo_evaluator/engine/gate.py [hunk 2]
```diff
@@ -340,6 +365,8 @@ def _metric_problem_values(
     """Return per-problem values for a supported gated metric."""
     by_problem: dict[int, list[dict[str, Any]]] = defaultdict(list)
     for (pid, _repeat), record in records.items():
+        if _is_infra_errored(record):
+            continue
```
[the remark this answers is not in the document]
Comment (arthi-arumugam-git) [reply src/nemo_evaluator/engine/gate.py]: You are right that the bundle fallback would bring the excluded zeros straight back. I did not filter `compare_runs` or recompute the bundle scores here, since those are computed over every sample by design and the regression report is only supporting evidence. What I did instead: when exclusions leave fewer than 10 pairs, the gate stays at INSUFFICIENT_EVIDENCE and adds a reason saying the bundle-level scores were not used, rather than falling back to them. There is a test for exactly that case (5 pairs left, bundle CIs that would BREACH on their own), and I checked that it fails with BREACH if the guard is removed. If you would rather have the fallback recompute from the filtered records, I can do that in a follow-up, it is a bigger change than this PR.

File: tests/test_gate.py [hunk 3]
```diff
@@ -488,8 +488,63 @@ def test_missing_metric_for_required_benchmark_fails_fast(self, tmp_path):
...
+        used to read as six wrong answers and turn a clean run into NO-GO.
+        """
+        base = _simple_records(20, 1.0)
+        cand = _simple_records(14, 1.0) + [
+            {
+                "problem_idx": i,
+                "repeat": 0,
+                "reward": 0.0,
+                "expected_answer": "ans",
+                "scoring_details": {"error": "sandbox unreachable", "error_category": "infra_error"},
```
[the remark this answers is not in the document]
Comment (arthi-arumugam-git) [reply tests/test_gate.py]: Done in ecd3ab5: parametrized over `infra_error` and `system`, added a case with `excluded_error_categories: []` in the policy (gates on every sample and goes NO-GO), and the sub-minimum boundary case that checks the gate does not BREACH on the bundle CI. Pulled the errored-record dict into a small helper so the tests do not repeat it.
